src/client/Microsoft.Identity.Client/Platforms/netcore/NetCorePoPCryptoProvider.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Security.Cryptography;
+using Microsoft.Identity.Client.AuthScheme.PoP;
+using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.Platforms.netcore
+{
+    /// <summary>
+    /// This provider is based on the RSACryptoServiceProvider container concept. 
+    /// It will create and store a pair of keys in the context of the current user. Subsequent calls to this provider will reuse this key.
+    /// </summary>
+    /// <remarks>
+    /// Key creation and storage only works on Windows. See https://stackoverflow.com/questions/41986995/implement-rsa-in-net-core/42006084 for more details.
+    /// </remarks>
+    internal class NetCorePoPCryptoProvider : IPoPCryptoProvider, IDisposable
```

Comment:
We discussed offline about persisting the key to disk or not. I believe now that is a bit too complex as there are too many issues in the code below (e.g. you persist the key even s_PersistKey is set to false). Since we do not need this functionality now, let's focus on quality and speed and abandon this persistence layer for now (which should not be used anyway) and favour the in-memory storage with expiration.